dsgrid/dimension/time.py

```diff
@@ -231,39 +284,291 @@ def is_standard(self):
             TimeZone.MST,
             TimeZone.CST,
             TimeZone.EST,
+            TimeZone.ARIZONA,
         ]
         if self in lst:
             return True
         return False
 
     def is_prevailing(self):
-        lst = [TimeZone.APT, TimeZone.PPT, TimeZone.MPT, TimeZone.CPT, TimeZone.EPT]
+        lst = [
+            TimeZone.APT,
+            TimeZone.PPT,
+            TimeZone.MPT,
+            TimeZone.CPT,
+            TimeZone.EPT,
+            TimeZone.ARIZONA,
+        ]
         if self in lst:
             return True
         return False
 
 
+def get_dls_springforward_time_change_by_year(
+    year: Union[int, list[int]], time_zone: TimeZone
```

Comment:
Can I keep both, list is more versatile while int is easier to use. They are not being used right now but there may be a use case at some point. For now we don't need to know when spring forward is because the drop is automatic with tz-aware time, we only need to know the fall back hour to be able to duplicate for interpolation.

Reply:
My general comment is that accepting multiple types for a function parameter is not a good practice because it adds unnecessary complexity. That behavior can usually be achieved more cleanly with multiple functions.

Everything about this function is simpler if you only accept an int, especially the return type (a single timestamp). As it is, even though the function name ends with "by_year", the user has to correlate the returned list with the years passed in.

If you want to keep the multiple option, here's what I suggest:
- Make one method that accepts a list and one that accepts an int. The latter calls the former.
- Change the return type of the "by_year" function to be a dictionary of year to timestamp.